components/feature/pwa/src/main/java/mozilla/components/feature/pwa/feature/WebAppHideToolbarFeature.kt

```diff
@@ -20,78 +39,78 @@ import mozilla.components.support.base.feature.LifecycleAwareFeature
  * Once the [Session] navigates to another scope, the toolbar will be revealed.
  *
  * @param toolbar Toolbar to show or hide.
- * @param sessionId ID of the custom tab session.
- * @param trustedScopes Scopes to hide the toolbar at.
- * Scopes correspond to [WebAppManifest.scope]. They can be a path (PWA) or just an origin (TWA).
- * @param onToolbarVisibilityChange Called when the toolbar is changed to be visible or hidden.
+ * @param store Reference to the browser store where tab state is located.
+ * @param customTabsStore Reference to the store that communicates with the custom tabs service.
+ * @param tabId ID of the tab session.
+ * @param manifest Reference to the cached [WebAppManifest] for the current PWA.
  */
 class WebAppHideToolbarFeature(
-    private val sessionManager: SessionManager,
     private val toolbar: View,
-    private val sessionId: String,
-    private var trustedScopes: List<Uri>,
-    private val onToolbarVisibilityChange: (visible: Boolean) -> Unit = {}
-) : Session.Observer, LifecycleAwareFeature {
+    private val store: BrowserStore,
+    private val customTabsStore: CustomTabsServiceStore? = null,
```

Comment:
`manifest` will be `null` if you're not using this feature for a PWA? Would it be better to split PWA-related functionality into a feature which extends this class?

Why are `customTabStore` and `tabId` nullable? At the very least I'd expect `tabId` to be not-null.

Reply:
`customTabStore` is null for backwards-compatibility. `tabId` is null to support working with the selected session, which is fairly common in Fenix.

Reply:
We can make a non-PWA/customtab version of this feature, but there isn't much logic to share. The store subscription code would be different if we're only listening to BrowserStore. I can add a separate version of this feature if we want.

Reply:
Okay, thanks. Can you expand docstrings to explain these choices?

Reply:
👍